src/main/java/net/sourceforge/plantuml/servlet/DiagramResponse.java

```diff
@@ -121,6 +121,10 @@ public static void init() {
             return;
         }
         initialized = true;
+        // set headless mode manually since otherwise Windows 11 seems to have some issues with it
+        // see Issue#311 :: https://github.com/plantuml/plantuml-server/issues/311
+        // NOTE: This can only be set before any awt/X11/... related stuff is loaded
+        System.setProperty("java.awt.headless", "true");
```

Comment:
An alternatve would be not to force it but only set it by default:
```java
System.setProperty("java.awt.headless", System.getProperty("java.awt.headless", "true"));
```

Maybe that would be a better solution? Should I change that?
The only way to force the PlantUML server to **not** run headless mode (for whatever reason) would now be to use the environment variable `PLANTUML_PROPERTY_FILE` like in this example : https://github.com/plantuml/plantuml-server/issues/311#issuecomment-1703229440

Reply:
I decided not to force the server to run in headless mode and still respect the users decision and only set the default/fallback to be `true` (headless mode). If the error still remains we can still adjust it.